library/ssl_tls13_generic.c

```diff
@@ -660,8 +708,21 @@ static int ssl_tls13_validate_certificate( mbedtls_ssl_context *ssl )
             ret = MBEDTLS_ERR_SSL_BAD_CERTIFICATE;
     }
 
+    /* mbedtls_x509_crt_verify_with_profile is supposed to report a
+     * verification failure through MBEDTLS_ERR_X509_CERT_VERIFY_FAILED,
```

Comment:
What about MBEDTLS_ERR_SSL_BAD_CERTIFICATE ?

Reply:
Seems mbedtls_x509_crt_verify_with_profile() will return MBEDTLS_ERR_X509_CERT_VERIFY_FAILED in case error happen. should we change it to MBEDTLS_ERR_SSL_BAD_CERTIFICATE ?